src/io/opendss.jl

```diff
@@ -968,22 +1418,36 @@ function parse_opendss(dss_data::Dict; import_all::Bool=false, vmin::Float64=0.9
     dss2tppm_shunt!(tppm_data, dss_data, import_all)
     dss2tppm_branch!(tppm_data, dss_data, import_all)
     dss2tppm_transformer!(tppm_data, dss_data, import_all)
+    dss2tppm_reactor!(tppm_data, dss_data, import_all)
     dss2tppm_gen!(tppm_data, dss_data, import_all)
     dss2tppm_pvsystem!(tppm_data, dss_data, import_all)
     dss2tppm_storage!(tppm_data, dss_data, import_all)
 
-    adjust_sourcegen_bounds!(tppm_data)
-
     tppm_data["dcline"] = []
 
     InfrastructureModels.arrays_to_dicts!(tppm_data)
+    # empty branch arrays are not converted into dicts;
+    # resolve this here to prevent issues later on
+    for key in ["branch"]
+        if isa(tppm_data[key], Array)
+            tppm_data[key] = Dict{String, Any}()
+        end
+    end
 
     for optional in ["dcline", "load", "shunt", "storage", "pvsystem"]
         if length(tppm_data[optional]) == 0
             tppm_data[optional] = Dict{String,Any}()
         end
     end
 
```

Comment:
This loop to turn empty `branch` arrays into a dict can be combined with the loop on line 1437, just add branch to the `optional` list.